sw/host/tests/rom/e2e_bootstrap_rma/src/main.rs

```diff
@@ -18,362 +21,322 @@ use opentitanlib::dif::lc_ctrl::{
+    let exit_failure = Regex::new("LCV:[0-9a-f]+\r\n").context("failed to build regex")?;
 
-    // The `exit_success` and `exit_failure` patterns are tightly coupled with
-    // sw/device/silicon_creator/rom/e2e/rom_e2e_bootstrap_rma_test.c.
-    let expected_bitfield = u32::from(DifRstmgrResetInfo::Por) | u32::from(DifRstmgrResetInfo::Sw);
     let mut console = UartConsole {
         timeout: Some(CONSOLE_TIMEOUT),
-        exit_success: Some(Regex::new(&format!(
-            "reset_info_bitfield: 0x{expected_bitfield:x}\r\n"
-        ))?),
-        exit_failure: Some(Regex::new(r"reset_info_bitfield: 0x[0-9a-f]+\r\n")?),
+        exit_success: Some(exit_success),
+        exit_failure: Some(exit_failure),
         ..Default::default()
     };
-    let result = console.interact(&*uart, None, Some(&mut std::io::stdout()))?;
-    if result != ExitStatus::ExitSuccess {
-        bail!("FAIL: {:?}", result);
-    };
-    Ok(())
-}
 
-fn main() -> Result<()> {
-    let opts = Opts::from_args();
-    opts.init.init_logging();
-    let transport = opts.init.init_target()?;
-    execute_test!(test_no_rma_command, &opts, &transport);
-    execute_test!(test_rma_command, &opts, &transport);
-    Ok(())
-}
+    let mut stdout = io::stdout();
+    let result = console
+        .interact(&*uart, None, Some(&mut stdout))
+        .context("failed to interact with console")?;
 
-#[cfg(test)]
-mod tests {
-    use super::*;
-
-    #[test]
-    fn openocd_statements() {
-        assert_eq!(
-            OpenocdTclBlock::AssertRegEq(LcCtrlReg::LcIdState, 0x3c).into_tcl(),
-            r#"
-set reg_value [ lc_ctrl.tap.0 riscv dmi_read 0xf ]
-if { $reg_value != 60 } {
-    echo "Expected LcIdState == 0x3c, but it is $reg_value"
-    nonexistent_command_that_causes_openocd_to_exit_with_error
-}"#
-        );
-
-        assert_eq!(
-            OpenocdTclBlock::PollUntilRegEq(LcCtrlReg::TransitionToken0, 0x1234).into_tcl(),
-            r#"
-for { set i 0 } { $i < 1000 } { set i [expr {$i + 1}] } {
-    set reg_value [ lc_ctrl.tap.0 riscv dmi_read 0x6 ]
-    if { $reg_value == 4660 } {
-        break
+    match result {
+        ExitStatus::ExitSuccess => Ok(()),
+        result => bail!("FAIL: {result:?}"),
     }
-    sleep 5
 }
-if { $i == 1000 } {
-    echo "Ran out of attempts before TransitionToken0 became 0x1234"
-    nonexistent_command_that_causes_openocd_to_exit_with_error
-}"#
-        );
-        assert_eq!(
-            OpenocdTclBlock::WriteReg(LcCtrlReg::TransitionToken3, 0xabcd).into_tcl(),
-            "lc_ctrl.tap.0 riscv dmi_write 0x9 0xabcd"
-        );
-
-        assert_eq!(OpenocdTclBlock::Echo("bar").into_tcl(), "echo \"bar\"");
+
+/// Reset the chip with a specified strapping configuration.
+///
+/// The given `strapping_confs` are names of known strapping configurations.
+/// See [`TransportWrapper::pin_strapping`] for details.
+fn reset_with_strapping(
```

Comment:
this function seems like overkill since you only use it once, can't you just:
1. apply strapping
2. transport.reset_target()
3. remove strapping (if you want to, or hold it if its a TAP strap)

Reply:
Yeah, I'd forgotten to remove it from this test. Thanks for catching!